internal/cli/atlas/search/nodes/create.go

```diff
@@ -64,10 +64,15 @@ func (opts *CreateOpts) Run() error {
 	}
 
 	if opts.EnableWatch {
-		if err := opts.Watch(opts.watcher); err != nil {
+		if err = opts.Watch(opts.watcher); err != nil {
 			return err
 		}
 		opts.Template = createWatchTemplate
+
+		r, err = opts.store.SearchNodes(opts.ConfigProjectID(), opts.clusterName)
+		if err != nil {
+			return err
+		}
```

Comment:
why do we need this extra check if nodes exist?
The watcher does that at line 82, and also checks for the stateName.
The watched will exit either if the stateName is IDLE, there is an error or on timeout.

Reply:
It's not an extra check if it exists, it's assigning the latest version to `r`.
The original version contains the state with the value `UPDATING`.

Reply:
got it, thanks for the explanations.
Might be a better option to update the watch framework, to return the entity watched too.
e.g. `func (opts *WatchOpts) Watch(f Watcher) error` => `func (opts *WatchOpts) Watch(f Watcher) (interface{}, error)` and `type Watcher func() (bool, error)` => `type Watcher func() (interface{}, bool, error)` (or generics [T any]).
This will avoid a redundant API call and also benefit us in the future too, when using watchers.
WDYT?

Reply:
Addressed in #2665, will rebase when it's merged

Reply:
LGTM, thank you